forms-flow-api/src/formsflow_api/resources/influenzaWorksite.py

```diff
@@ -124,3 +124,21 @@ def get():
         except BusinessException as err:
             current_app.logger.warning(err.error)
             return err.error, err.status_code
+
+@cors_preflight("DELETE, OPTIONS")
+@API.route("/worksites_registrations/<int:application_id>", methods=["DELETE", "OPTIONS"])
+class InfluenzaWorksiteById(Resource):
+    """Resource for worksites_registrations by id"""
+
+    @staticmethod
+    @profiletime
+    @auth.require
+    @auth.has_one_of_roles(["formsflow-designer"])
+    def delete(application_id: int):
+        """delete influenza worksites_registrations by application_id"""
+        try:
+            InfluenzaService.delete_worksites_registrations(application_id)
+            return "Deleted", HTTPStatus.OK
+        except BusinessException as err:
+            current_app.logger.warning(err.error)
+            return err.error, err.status_code
```

Comment:
Enabling a designer to directly remove an application in ODS warehouse. Not an exact requirement, added just in case 🙃

Reply:
@iman-jamali-fw user will be able to delete their own form. does this decorator allow that too?

Reply:
Not from this endpoint. 

There is another endpoint in the `application` resource that will allow the users to delete their own submissions (remove it from all related data sources). This one is just in case the designer wants to delete it from ods directly.